zerver/data_import/slack_message_conversion.py

```diff
@@ -157,13 +157,19 @@ def convert_markdown_syntax(text: str, regex: str, zulip_keyword: str) -> str:
 def convert_link_format(text: str) -> tuple[str, bool]:
     """
     1. Converts '<https://foo.com>' to 'https://foo.com'
-    2. Converts '<https://foo.com|foo>' to 'https://foo.com|foo'
+    2. Converts '<https://foo.com|foo>' to '[foo](https://foo.com)'
     """
     has_link = False
     for match in re.finditer(LINK_REGEX, text, re.VERBOSE):
-        converted_text = match.group(0).replace(">", "").replace("<", "")
+        slack_url = match.group(0)
+        base_url = slack_url[1:-1]  # remove "<" & ">"
+        url_parts = base_url.split("|")
+        converted_url = url_parts[0]
+        if 1 < len(url_parts) < 3:  # Check if there's a pipe with text after it
```

Comment:
Is that possible with the link regex? If so, then you could set the [`max_split` parameter](https://docs.python.org/3/library/stdtypes.html#str.split) to one.

```
url_parts = base_url.split("|", 1)
```

Reply:
It can, I added this fixture in `slack_message_conversion.json` to test that case:
```json
    {
      "name": "slack_link_with_pipes",
      "input": ">Google logo today:\n><https://foo.com|foo|oof>\n>Kinda boring",
      "conversion_output": ">Google logo today:\n>[foo|oof](https://foo.com)\n>Kinda boring"
    },
```

Reply:
And, tidied up the logic:
```diff
diff --git a/zerver/data_import/slack_message_conversion.py b/zerver/data_import/slack_message_conversion.py
index e9375128ee..f35363e019 100644
--- a/zerver/data_import/slack_message_conversion.py
+++ b/zerver/data_import/slack_message_conversion.py
@@ -162,10 +162,10 @@ def convert_link_format(text: str) -> tuple[str, bool]:
     has_link = False
     for match in re.finditer(LINK_REGEX, text, re.VERBOSE):
         slack_url = match.group(0)
-        base_url = slack_url[1:-1]  # remove "<" & ">"
-        url_parts = base_url.split("|")
+        url_parts = slack_url[1:-1].split("|", maxsplit=1)
         converted_url = url_parts[0]
-        if 1 < len(url_parts) < 3:  # Check if there's a pipe with text after it
+        # Check if there's a pipe with text after it
+        if len(url_parts) == 2:
             converted_url = f"[{url_parts[1]}]({url_parts[0]})"
 
         has_link = True
```